### DISCUSSION TITLE
Stepper: the lock behavior is a bit confusing

### DISCUSSION THREAD

**Discussion by Etchelon** [skeletonlabs/skeleton#774] [Ideas]
Hello, I am using the Stepper component and I'm finding the lock behavior to be a bit confusing:

<img width="70" alt="image" src="https://user-images.githubusercontent.com/4141889/211009483-4e2cd1ba-823a-4a22-9d5d-3946aea03f9f.png">

As you see in the screenshot, the active step is the first, and the next one, which has a `locked = {check validity of 2nd step}` shows a locked icon.
Just by the look of things, it seems the next step is not available until I complete the first one, but in reality the lock refers to the validity of the 2nd one.

What about:
- showing the lock on a step only when it's active?
- or rather, changing the behavior altogether and mark all steps after a "locked" one to show the lock until that one step is valid? I think it's more natural.

Think of any videogame: until you complete a level, the next ones are locked (the lock doesn't show on the current one!).

**Comment by endigo9740** [comment]
Currently that state is tracked within the step itself, so I'm not sure if we can pass that to the next step. But we have some ideas in mind to rework with Stepper. I'll take this feedback into consideration. Thanks for pointing this out!
